### PR TITLE
docs(contributing): add guides on how to contribute

### PR DESCRIPTION
**Pull request by tthvo**
Related to #580 

Added some simple steps on how to contribute to Cryostat projects.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (andrewazores) [inline CONTRIBUTING.md:31]: `merge` and `rebase` are different, best to have the command and the description match up. For the purpose of opening a PR either is fine, though personally I prefer to rebase.

File: CONTRIBUTING.md
Comment (andrewazores) [inline CONTRIBUTING.md:50]: "tag" -> "label" ?

File: CONTRIBUTING.md
Comment (andrewazores) [inline CONTRIBUTING.md:51]: I'd remove this item. This is really the maintainer's concern, not the individual contributor.

File: CONTRIBUTING.md
Comment (andrewazores) [inline CONTRIBUTING.md:52]: on -> one

File: CONTRIBUTING.md
Comment (andrewazores) [inline CONTRIBUTING.md:68]: "- before deleting the current checked out one" ?

File: CONTRIBUTING.md
Comment (andrewazores) [inline CONTRIBUTING.md:69]: `git branch -D` ?

Maybe the comment should specify "local branch"

File: CONTRIBUTING.md
Comment (tthvo) [reply CONTRIBUTING.md:31]: Right! Fixed that!

File: CONTRIBUTING.md
Comment (tthvo) [reply CONTRIBUTING.md:69]: Right makes sense! Must have been fast-typed :D Fixed it :D

**Comment by andrewazores** [comment]
Doc looks good, change to `web-client` submodule must be dropped.

Was this feature branch merged or rebased? :-P

**Comment by tthvo** [comment]
> Was this feature branch merged or rebased? :-P

Not sure what it means here... :D

File: CONTRIBUTING.md
Comment (maxcao13) [inline CONTRIBUTING.md:33]: I think you can just 
```
To rebase: 
$ git fetch upstream
$ git rebase upstream/main
```
```
To push:
$ git push -f
```

I see that this you are rebasing your own fork first, so maybe mention that's what is happening? Like
```
To rebase your fork with the latest changes: 
...
...
# Checkout main, rebase on top of upstream, pull from origin
```
Or something like that.

File: CONTRIBUTING.md
Comment (maxcao13) [inline CONTRIBUTING.md:1]: Not sure if it should be done here or in another file/pr, but I would mention how to structure the pull request titles, and labels, and such.

File: CONTRIBUTING.md
Comment (tthvo) [reply CONTRIBUTING.md:33]: Make sense to me! I will fix this.

File: CONTRIBUTING.md
Comment (tthvo) [reply CONTRIBUTING.md:1]: Not sure but this documentation in this file seems to explains pretty well: https://www.conventionalcommits.org/en/v1.0.0/

Comment (tthvo) [reply CONTRIBUTING.md:1]: As for label, do you think they are self-explained enough tho :d?

Comment (maxcao13) [reply CONTRIBUTING.md:1]: Sure I guess that part makes sense not to include.

**Review by andrewazores — COMMENTED** [review]
LGTM

File: CONTRIBUTING.md
Comment (cmah88) [inline CONTRIBUTING.md:11]: update references to cryostat instead of cryostatio

the below worked for me:
git clone https://github.com/<username>/cryostat.git

File: CONTRIBUTING.md
Comment (cmah88) [inline CONTRIBUTING.md:15]: is this checking out a new branch? suggest "Navigate to Cryostat directory and checkout new branch" if so

File: CONTRIBUTING.md
Comment (cmah88) [inline CONTRIBUTING.md:23]: this message seems to become the title- maybe add title guidelines here? (A title following [guidelines](https://www.conventionalcommits.org/en/v1.0.0/).)

File: CONTRIBUTING.md
Comment (tthvo) [reply CONTRIBUTING.md:11]: Since we are using our `cryostatio.github.io.git` as an example, do you think we should keep as current?

File: CONTRIBUTING.md
Comment (tthvo) [reply CONTRIBUTING.md:23]: Right! I think eventually you can edit the title but it makes sense to suggest it here.

File: CONTRIBUTING.md
Comment (tthvo) [reply CONTRIBUTING.md:11]: > Since we are using our `cryostatio.github.io.git` as an example, do you think we should keep as current?

I think maybe I could find a PR in `cryostat` and replace the screenshots too. That would be easier since cryostat is the main.

**Comment by tthvo** [comment]
Addressed new suggestions and updated with format following here: https://generator.contributing.md/#

**Review by maxcao13 — DISMISSED** [review]
LGTM

**Comment by andrewazores** [comment]
Before I merge this, could you actually add a brief note near the top of the `README` to point out this new doc? Just to highlight that it's there since the `README` is the first thing a user sees when they land on the repo, and it's much more prominent and noticeable than the plain file list.

**Review by andrewazores — APPROVED** [review]

**Pull request merged**
